cmd/erasure-object.go

```diff
@@ -1047,7 +1047,7 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 		if opts.VersionPurgeStatus.Empty() && opts.DeleteMarkerReplicationStatus == "" {
 			markDelete = false
 		}
-		if opts.DeleteMarker && opts.VersionPurgeStatus == Complete {
```

Comment:
We should get rid of this as well

```
deleteMarker := true
if gerr == nil {
   deleteMarker = goi.VersionID != ""
}
```

We can remove this silly code few lines above as well, its all `deleteMarker` is always `true`

Reply:
removed.